gobblin-api/src/main/java/gobblin/source/extractor/WatermarkInterval.java

```diff
@@ -0,0 +1,91 @@
+package gobblin.source.extractor;
+
+import gobblin.fork.CopyNotSupportedException;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import com.google.common.base.Preconditions;
+
+public class WatermarkInterval implements Writable {
+
+  private Watermark lowWatermark;
+  private Watermark expectedHighWatermark;
+  private Watermark actualHighWatermark;
+
+  private ObjectMapper objectMapper;
+
+  // Needed for the Writable interface
+  public WatermarkInterval() {
+  }
+
+  private WatermarkInterval(Watermark lowWatermark, Watermark expectedHighWatermark) throws CopyNotSupportedException {
+    this.lowWatermark = lowWatermark;
+    this.expectedHighWatermark = expectedHighWatermark;
+    this.actualHighWatermark = lowWatermark.copy();
```

Comment:
I suggest having a protected abstract method for initializing the `actualHighWatermark` instead of requiring an `Watermark` to be a `Copyable`. Then implementation classes can provide the right behavior for initializing it.